core/cas-server-core-authentication/src/main/java/org/apereo/cas/authentication/DefaultAuthenticationEventExecutionPlan.java

```diff
@@ -71,7 +71,7 @@ public void registerMetadataPopulators(final Collection<AuthenticationMetaDataPo
     public Set<AuthenticationHandler> getAuthenticationHandlersForTransaction(final AuthenticationTransaction transaction) {
         final AuthenticationHandler[] handlers = authenticationHandlerPrincipalResolverMap.keySet().toArray(new AuthenticationHandler[]{});
         OrderComparator.sortIfNecessary(handlers);
-        return new LinkedHashSet<>(Arrays.stream(handlers).collect(Collectors.toSet()));
+        return new LinkedHashSet<>(Arrays.asList(handlers));
```

Comment:
Please review the relevant section in the master branch and backport accordingly. Thank you.